iOS/FVM/Framework/GestureHandlers/TapGestureHandler.swift

```diff
@@ -20,16 +21,24 @@ extension FVMCarModelViewController {
         if hitResults.count > 0 {
             let result = hitResults.first!
             guard let nodeName = result.node.name else {
-                print("Can't highlight part without name")
+                Log.warning("Can't highlight part without name")
                 return
             }
             if (highlightHandler.isHighlighted(nodeName: nodeName)) {
-                highlightHandler.setHighlightOff(nodeName: nodeName)
+                setHighlightOff(nodeName: nodeName)
             } else {
-                highlightHandler.setHighlightOn(node: result.node)
+                setHighlightOn(node: result.node, nodeName: nodeName)
             }
-        } else {
-            highlightHandler.setAllHighlightsOff()
         }
     }
+    
+    private func setHighlightOn(node: SCNNode, nodeName: String) {
+        damagedPartsService.addPart(part: Selection(newName: nodeName))
```

Comment:
You have a property **name** in SCNode. You don't need paste nodeName here.

Reply:
right, changed